diff --git a/launcher/LaunchController.cpp b/launcher/LaunchController.cpp
index d33dbc3f1..9be44fe15 100644
--- a/launcher/LaunchController.cpp
+++ b/launcher/LaunchController.cpp
@@ -213,8 +213,9 @@ void LaunchController::login()
     if ((m_accountToUse->accountType() != AccountType::Offline && m_accountToUse->accountState() == AccountState::Offline) ||
         m_accountToUse->shouldRefresh()) {
         // Force account refresh on the account used to launch the instance updating the AccountState
-        // only on first try and if it is not meant to be offline
-        m_accountToUse->refresh();
+        //  only on first try and if it is not meant to be offline
+        auto accounts = APPLICATION->accounts();
+        accounts->requestRefresh(m_accountToUse->internalId());
     }
     while (tryagain) {
         if (tries > 0 && tries % 3 == 0) {
@@ -233,34 +234,13 @@ void LaunchController::login()
         m_session->demo = m_demo;
         m_accountToUse->fillSession(m_session);
 
-        MinecraftAccountPtr accountToCheck;
-
-        if (m_accountToUse->ownsMinecraft())
-            accountToCheck = m_accountToUse;
-        else if (const MinecraftAccountPtr defaultAccount = APPLICATION->accounts()->defaultAccount();
-                 defaultAccount != nullptr && defaultAccount->ownsMinecraft()) {
-            accountToCheck = defaultAccount;
-        } else {
-            for (int i = 0; i < APPLICATION->accounts()->count(); i++) {
-                MinecraftAccountPtr account = APPLICATION->accounts()->at(i);
-                if (account->ownsMinecraft())
-                    accountToCheck = account;
-            }
-        }
-
-        if (accountToCheck == nullptr) {
-            if (!m_session->demo)
-                m_session->demo = askPlayDemo();
-
-            if (m_session->demo)
-                launchInstance();
-            else
-                emitFailed(tr("Launch cancelled - account does not own Minecraft."));
-
+        // Launch immediately in true offline mode
+        if (m_accountToUse->accountType() == AccountType::Offline) {
+            launchInstance();
             return;
         }
 
-        switch (accountToCheck->accountState()) {
+        switch (m_accountToUse->accountState()) {
             case AccountState::Offline: {
                 m_session->wants_online = false;
             }
@@ -281,37 +261,23 @@ void LaunchController::login()
                     }
                     m_session->MakeOffline(name);
                     // offline flavored game from here :3
-                } else if (m_accountToUse == accountToCheck && !m_accountToUse->hasProfile()) {
-                    // Now handle setting up a profile name here...
-                    ProfileSetupDialog dialog(m_accountToUse, m_parentWidget);
-                    if (dialog.exec() == QDialog::Accepted) {
-                        tryagain = true;
-                        continue;
-                    } else {
-                        emitFailed(tr("Received undetermined session status during login."));
-                        return;
-                    }
                 }
-
-                if (m_accountToUse->accountType() == AccountType::Offline)
-                    m_session->wants_online = false;
-
-                // we own Minecraft, there is a profile, it's all ready to go!
                 launchInstance();
                 return;
             }
             case AccountState::Errored:
                 // This means some sort of soft error that we can fix with a refresh ... so let's refresh.
             case AccountState::Unchecked: {
-                accountToCheck->refresh();
+                m_accountToUse->refresh();
             }
             /* fallthrough */
             case AccountState::Working: {
                 // refresh is in progress, we need to wait for it to finish to proceed.
                 ProgressDialog progDialog(m_parentWidget);
-                progDialog.setSkipButton(true, tr("Abort"));
-
-                auto task = accountToCheck->currentTask();
+                if (m_online) {
+                    progDialog.setSkipButton(true, tr("Play Offline"));
+                }
+                auto task = m_accountToUse->currentTask();
                 progDialog.execWithTask(task.get());
 
                 // don't retry if aborted
@@ -321,14 +287,12 @@ void LaunchController::login()
                 continue;
             }
             case AccountState::Expired: {
-                if (reauthenticateAccount(accountToCheck))
+                if (reauthenticateCurrentAccount())
                     continue;
                 return;
             }
             case AccountState::Disabled: {
-                auto errorString = tr("The launcher's client identification has changed. Please remove '%1' and try again.")
-                                       .arg(accountToCheck->profileName());
-
+                auto errorString = tr("The launcher's client identification has changed. Please remove this account and add it again.");
                 QMessageBox::warning(m_parentWidget, tr("Client identification changed"), errorString, QMessageBox::StandardButton::Ok,
                                      QMessageBox::StandardButton::Ok);
                 emitFailed(errorString);
@@ -336,9 +300,8 @@ void LaunchController::login()
             }
             case AccountState::Gone: {
                 auto errorString =
-                    tr("'%1' no longer exists on the servers. It may have been migrated, in which case please add the new account "
-                       "you migrated this one to.")
-                        .arg(accountToCheck->profileName());
+                    tr("The account no longer exists on the servers. It may have been migrated, in which case please add the new account "
+                       "you migrated this one to.");
                 QMessageBox::warning(m_parentWidget, tr("Account gone"), errorString, QMessageBox::StandardButton::Ok,
                                      QMessageBox::StandardButton::Ok);
                 emitFailed(errorString);
@@ -349,35 +312,30 @@ void LaunchController::login()
     emitFailed(tr("Failed to launch."));
 }
 
-bool LaunchController::reauthenticateAccount(MinecraftAccountPtr account)
+bool LaunchController::reauthenticateCurrentAccount()
 {
-    auto button = QMessageBox::warning(
-        m_parentWidget, tr("Account refresh failed"),
-        tr("'%1' has expired and needs to be reauthenticated. Do you want to reauthenticate this account?").arg(account->profileName()),
-        QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::No, QMessageBox::StandardButton::Yes);
+    auto button =
+        QMessageBox::warning(m_parentWidget, tr("Account refresh failed"),
+                             tr("The account has expired and needs to be reauthenticated. Do you want to reauthenticate this account?"),
+                             QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::No, QMessageBox::StandardButton::Yes);
     if (button == QMessageBox::StandardButton::Yes) {
         auto accounts = APPLICATION->accounts();
-        bool isDefault = accounts->defaultAccount() == account;
-        accounts->removeAccount(accounts->index(accounts->findAccountByProfileId(account->profileId())));
-        if (account->accountType() == AccountType::MSA) {
+        bool isDefault = accounts->defaultAccount() == m_accountToUse;
+        accounts->removeAccount(accounts->index(accounts->findAccountByProfileId(m_accountToUse->profileId())));
+        if (m_accountToUse->accountType() == AccountType::MSA) {
             auto newAccount = MSALoginDialog::newAccount(m_parentWidget);
-
-            if (newAccount != nullptr) {
-                accounts->addAccount(newAccount);
-
-                if (isDefault)
-                    accounts->setDefaultAccount(newAccount);
-
-                if (m_accountToUse == account) {
-                    m_accountToUse = nullptr;
-                    decideAccount();
-                }
-                return true;
+            accounts->addAccount(newAccount);
+            if (isDefault) {
+                accounts->setDefaultAccount(newAccount);
             }
+            m_accountToUse = nullptr;
+            decideAccount();
+            return true;
         }
+        emitFailed(tr("Account expired and re-login attempt failed"));
+    } else {
+        emitFailed(tr("The account has expired and needs to be reauthenticated"));
     }
-
-    emitFailed(tr("The account has expired and needs to be reauthenticated"));
     return false;
 }
 
diff --git a/launcher/LaunchController.h b/launcher/LaunchController.h
index 50b72eb18..b0e2271a2 100644
--- a/launcher/LaunchController.h
+++ b/launcher/LaunchController.h
@@ -77,8 +77,8 @@ class LaunchController : public Task {
     void launchInstance();
     void decideAccount();
     bool askPlayDemo();
-    QString askOfflineName(QString playerName, bool demo, bool* ok = nullptr);
-    bool reauthenticateAccount(MinecraftAccountPtr account);
+    QString askOfflineName(QString playerName, bool demo, bool* ok);
+    bool reauthenticateCurrentAccount();
 
    private slots:
     void readyForLaunch();
diff --git a/launcher/minecraft/auth/AccountList.cpp b/launcher/minecraft/auth/AccountList.cpp
index cef79d200..57a0cf7ca 100644
--- a/launcher/minecraft/auth/AccountList.cpp
+++ b/launcher/minecraft/auth/AccountList.cpp
@@ -561,12 +561,7 @@ void AccountList::setListFilePath(QString path, bool autosave)
 
 bool AccountList::anyAccountIsValid()
 {
-    for (auto account : m_accounts) {
-        if (account->ownsMinecraft()) {
-            return true;
-        }
-    }
-    return false;
+    return true;
 }
 
 void AccountList::fillQueue()
