diff --git a/bottles/frontend/meson.build b/bottles/frontend/meson.build
index 6ff7c01..c26ea0b 100644
--- a/bottles/frontend/meson.build
+++ b/bottles/frontend/meson.build
@@ -23,12 +23,6 @@ params_file = configure_file(
     configuration: conf
 )
 
-fs = import('fs')
-
-if not fs.is_file('/' + '.flatpak-info')
-  error('file does not exist')
-endif
-
 bottles_sources = [
   '__init__.py',
   'main.py',
diff --git a/bottles/frontend/utils/autostart.py b/bottles/frontend/utils/autostart.py
index 369f964..acb73fb 100644
--- a/bottles/frontend/utils/autostart.py
+++ b/bottles/frontend/utils/autostart.py
@@ -21,7 +21,7 @@ def set_autostart_enabled(
     sandboxed=None,
 ):
     if sandboxed is None:
-        sandboxed = Xdp.Portal.running_under_sandbox()
+        sandboxed = self.window.settings.get_boolean("show-sandbox-warning")
 
     if not sandboxed:
         callback(ManagerUtils.set_autostart_entry(enabled))
diff --git a/bottles/frontend/views/bottle_details.py b/bottles/frontend/views/bottle_details.py
index dfeec4e..300978f 100644
--- a/bottles/frontend/views/bottle_details.py
+++ b/bottles/frontend/views/bottle_details.py
@@ -21,7 +21,7 @@ from datetime import datetime
 from gettext import gettext as _
 from typing import Dict, List, Optional
 
-from gi.repository import Adw, Gdk, Gio, GLib, Gtk, Xdp
+from gi.repository import Adw, Gdk, Gio, GLib, Gtk
 
 from bottles.backend.managers.backup import BackupManager
 from bottles.backend.models.config import BottleConfig
@@ -727,20 +727,17 @@ class BottleView(Adw.PreferencesPage):
             dialog.connect("response", execute)
             dialog.show()
 
-        if Xdp.Portal.running_under_sandbox():
-            if self.window.settings.get_boolean("show-sandbox-warning"):
-                dialog = Adw.MessageDialog.new(
-                    self.window,
-                    _("Be Aware of Sandbox"),
-                    _(
-                        "Bottles is running in a sandbox, a restricted permission environment needed to keep you safe. If the program won't run, consider moving inside the bottle (3 dots icon on the top), then launch from there."
-                    ),
-                )
-                dialog.add_response("dismiss", _("_Dismiss"))
-                dialog.connect("response", lambda *args: show_chooser())
-                dialog.present()
-            else:
-                show_chooser()
+        if self.window.settings.get_boolean("show-sandbox-warning"):
+            dialog = Adw.MessageDialog.new(
+                self.window,
+                _("Be Aware of Sandbox"),
+                _(
+                    "Bottles is running in a sandbox, a restricted permission environment needed to keep you safe. If the program won't run, consider moving inside the bottle (3 dots icon on the top), then launch from there."
+                ),
+            )
+            dialog.add_response("dismiss", _("_Dismiss"))
+            dialog.connect("response", lambda *args: show_chooser())
+            dialog.present()
         else:
             show_chooser()
 
diff --git a/bottles/frontend/views/bottle_preferences.py b/bottles/frontend/views/bottle_preferences.py
index 0254eb9..d57e919 100644
--- a/bottles/frontend/views/bottle_preferences.py
+++ b/bottles/frontend/views/bottle_preferences.py
@@ -20,7 +20,7 @@ import os
 import re
 from gettext import gettext as _
 
-from gi.repository import Adw, Gdk, Gtk, Xdp
+from gi.repository import Adw, Gdk, Gtk
 
 from bottles.backend.globals import (
     gamemode_available,
@@ -185,7 +185,7 @@ class PreferencesView(Adw.PreferencesPage):
                 )
             )
 
-        if not is_cpak() and not Xdp.Portal.running_under_sandbox():
+        if not is_cpak() and not self.window.settings.get_boolean("show-sandbox-warning"):
             return
 
         _not_available = _("This feature is unavailable on your system.")
diff --git a/bottles/frontend/views/list.py b/bottles/frontend/views/list.py
index 689d023..6e437ba 100644
--- a/bottles/frontend/views/list.py
+++ b/bottles/frontend/views/list.py
@@ -19,7 +19,7 @@ from datetime import datetime
 from gettext import gettext as _
 from gettext import ngettext
 
-from gi.repository import Adw, GLib, Gtk, Xdp
+from gi.repository import Adw, GLib, Gtk
 
 from bottles.backend.globals import is_cpak
 from bottles.backend.models.config import BottleConfig
@@ -130,9 +130,6 @@ class BottlesBottleRow(Adw.ActionRow):
 
     def run_executable(self, *_args):
         """Display file dialog for executable"""
-        if not is_cpak() and not Xdp.Portal.running_under_sandbox():
-            return
-
         def set_path(_dialog, response):
             if response != Gtk.ResponseType.ACCEPT:
                 return
diff --git a/bottles/frontend/views/new_bottle_dialog.py b/bottles/frontend/views/new_bottle_dialog.py
index 02adeb2..cf3e938 100644
--- a/bottles/frontend/views/new_bottle_dialog.py
+++ b/bottles/frontend/views/new_bottle_dialog.py
@@ -20,7 +20,7 @@ from gettext import gettext as _
 from threading import Event
 from typing import Any, Optional
 
-from gi.repository import Adw, Gio, GLib, GObject, Gtk, Pango, Xdp
+from gi.repository import Adw, Gio, GLib, GObject, Gtk, Pango
 from pathvalidate import sanitize_filename
 
 from bottles.backend.globals import is_cpak
@@ -93,9 +93,7 @@ class BottlesNewBottleDialog(Adw.Dialog):
         )
         # common variables and references
         self.window = GtkUtils.get_parent_window()
-        if not self.window or (
-            not is_cpak() and not Xdp.Portal.running_under_sandbox()
-        ):
+        if not self.window or not is_cpak():
             return
 
         self.app = self.window.get_application()
diff --git a/bottles/frontend/windows/window.py b/bottles/frontend/windows/window.py
index 146a99e..d601630 100644
--- a/bottles/frontend/windows/window.py
+++ b/bottles/frontend/windows/window.py
@@ -21,7 +21,7 @@ import webbrowser
 from datetime import datetime, timedelta
 from gettext import gettext as _
 
-from gi.repository import Adw, Gdk, Gio, GLib, Gtk, Xdp, XdpGtk4
+from gi.repository import Adw, Gdk, Gio, GLib, Gtk, XdpGtk4
 
 from bottles.backend.globals import Paths
 from bottles.backend.health import HealthChecker
@@ -138,35 +138,6 @@ class BottlesWindow(Adw.ApplicationWindow):
             manager = Adw.StyleManager.get_default()
             manager.set_color_scheme(Adw.ColorScheme.FORCE_DARK)
 
-        # Be VERY explicit that non-sandboxed environments are unsupported
-        if not os.environ.get("CPAK_CONTAINER_ID") and not Xdp.Portal.running_under_sandbox():
-
-            def response(dialog, response, *args):
-                if response == "close":
-                    quit(1)
-
-            body = _(
-                "Bottles is only supported within a sandboxed environment. Official sources of Bottles are available at"
-            )
-            download_url = "usebottles.com/download"
-
-            error_dialog = Adw.AlertDialog.new(
-                _("Unsupported Environment"),
-                f"{body} <a href='https://{download_url}' title='https://{download_url}'>{download_url}.</a>",
-            )
-
-            error_dialog.add_response("close", _("Close"))
-            error_dialog.set_body_use_markup(True)
-            error_dialog.connect("response", response)
-            error_dialog.present(self)
-            logging.error(
-                _(
-                    "Bottles is only supported within a sandboxed format. Official sources of Bottles are available at:"
-                )
-            )
-            logging.error("https://usebottles.com/download/")
-            return
-
         # Loading view
         self.page_loading = LoadingView()
 
