From 2060bd8258223ce3b5c491fc2abf5b21c220f379 Mon Sep 17 00:00:00 2001
From: Naz <ndpm13@ch-naseem.com>
Date: Thu, 2 Apr 2026 09:53:19 +0100
Subject: [PATCH] refactor: change Linux build target to dir only

---
 _scripts/build.mjs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/_scripts/build.mjs b/_scripts/build.mjs
index 1617f1a..5720b79 100644
--- a/_scripts/build.mjs
+++ b/_scripts/build.mjs
@@ -34,7 +34,7 @@ if (platform === 'darwin') {
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', '7z', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const output = await build({ targets, config, publish: 'never' })
-- 
2.53.0

